Add Phison MSD210 (MSD210-M8320GPSCB5UFI-S122)

phison-someone opened the pull request:
To Whom It May Concern,

I am writing to submit a pull request to modify Phison's MSD210 series.
drivedb.h & change log have been updated.

Please let me know if you have any questions. Thank you.

phison-someone committed:
Update drivedb.h_MSD210

phison-someone committed:
Update ChangeLog

phison-someone committed:
Update ChangeLog

phison-someone committed:
Update ChangeLog

phison-someone committed:
Update ChangeLog

phison-someone committed:
Update ChangeLog

phison-someone committed:
Merge pull request #2 from phison-someone/phison-someone-patch-2-1

update change log

phison-someone committed:
Update ChangeLog

phison-someone committed:
Update drivedb.h

phison-someone committed:
Update drivedb.h

phison-someone committed:
Update drivedb.h

phison-someone committed:
Update ChangeLog

phison-someone committed:
Merge branch 'master' into phison-someone-patch-MSD210

phison-someone commented:
smartctrl

[logtest001 (2).log](https://github.com/user-attachments/files/20153332/logtest001.2.log)

chrfranke commented:
The [accumulated diff](./343.diff) of this PR shows that the model family string is missing. Smartctl refuses the patched file:
```
$ smartctl -B ./drivedb.h -P showall 'MSD210-M8320GPSCB5UFI-S122'
./drivedb.h(192): Syntax error, ',' expected
```
This makes the PR useless.

**Important: PLEASE test each new entry before posting as a PR.**

The attributes look like a subset of the existing entry `Phison Driven SSDs`. Please consider to add the new drives to the regex of the existing entry or explain why this would not work.

phison-someone committed:
Update drivedb.h

add model family info

phison-someone committed:
Update drivedb.h

modify Informal string about the model family.

phison-someone committed:
Update drivedb.h

1. below the existing Phison entries
2. delete "SPCC Solid State Disk"
3. modify firmware and model name regex

phison-someone commented:
> * the regex
Phison Driven SSDs :
"-v 173,raw16(avg16),MaxAvgErase_Ct "
Phison MSD210 Series SSD :
"-v 173,raw24/raw24:z10z32,Erase_Ct_Max/Avg "
The above two (173) are different. So we need add new entries.
Other parts have been modified.
Thanks

> Thanks.
> 
> * Please explain why an entry separate from `Phison Driven SSDs` entry is needed. We do not add new entries for each new drive series if the attribute settings are similar.
> * The new entry should be located below or above the existing `Phison ...` entries.
> * The `"SPCC Solid State Disk" // ... ` part is obviously block copied from `Phison Driven OEM SSDs` which does not make sense.
> * The firmware regex `S[89ABCH]F[DMI][0-9T][0-9]\\.[0-9]` is not needed unless  `"SPCC Solid State Disk"` is really intentional. In the latter case, please fix the block copied `// tested with ...` comment and provide a sample `smartctl -x -a` output for a `"SPCC Solid State Disk"` device.
> * In the regex, please use `[..]` instead of `(.|.)` for single char alternatives, for example `[48]` instead of `(4|8)`.
> * The ChangeLog comment should be informal only and should not contain the full regex.

chrfranke commented:
The 'master' branch is now retired, see #346. Please create a new PR for the 'main' branch.
Please also move the new entry to above or below the existing entry `Phison Driven SSDs`.
